byway/src/main/webapp/routepage.js

```diff
@@ -159,19 +164,39 @@ function clearRoute(){
 /** Get trip info from datastore onload */
 function getRouteOnload(){
   clearRoute();
-  fetch('/api/stop')
+  fetch(configureTripKeyForPath(tripKey, "/api/stop"))
   .then(response => response.json())
-  .then((trip) => {
+  .then(async (trip) => {
     if(trip != null){
-      start = end = trip.start;
-      destinations = trip.destinations;
-      route.push(...trip.route)
+      try{
+        let res = await findPlace(trip.start);
+        start = end = res;
+   
+      } catch (error) {
+        console.error("Could not retrieve a start nor end point due to: ", error);
+      }
+      
+      for(let destinationId of trip.destinations){
+        try{
+          let destinationAsPlaceObj = await findPlace(destinationId);
+          destinations.push(destinationAsPlaceObj);
+        } catch (error) {
+          console.error("Could not retrieve destinations due to: ", error);
+        }
+      }
 
+      for(let waypointId of trip.route){
+        try{
+          let waypointAsPlaceObj = await findPlace(waypointId);
+          route.push(waypointAsPlaceObj);
+        } catch (error) {
+          console.error("Could not retrieve route due to: ", error);
+        }
+      }
       calcRoute();
     }
     else{
-      console.log("Could not retrieve any routes nor destinations associated with this trip." +
-       "Please reload page and try again.");
+      console.log("Could not retrieve any routes nor destinations associated with this trip. Please reload page and try again.");
```

Comment:
In a future pull request, something worth considering is converting all these `console.log` and `alert()` messages to error messages show to the user in the HTML. It would be nice to have better error handling from a user perspective.

Reply:
Do you mean have a field to display on the HTML? Why is bad to have these alert messages instead? Will keep this in mind, thanks.